packages/ast-spec/src/expression/ClassExpression/spec.ts

```diff
@@ -5,5 +5,4 @@ export interface ClassExpression extends ClassBase {
   type: AST_NODE_TYPES.ClassExpression;
   abstract: false;
   declare: false;
-  decorators: [];
```

Comment:
This was added in 450dd47c0d35c54ae8ac693a4c67905df79488a2 as `decorators?: undefined` to demark that unlike `ClassBase` (and thus `ClassDeclaration`), `ClassExpression` had no decorators.
Then in df131e258c93e5714c88c0373cfeb2e1e75afc75 Josh removed the optionality from the tree and forced this to be an empty array.

And indeed you can see here that the converter always sets this to an empty array:
https://github.com/typescript-eslint/typescript-eslint/blob/988b31e3be10ad83018b20338db3aefa0c570a08/packages/typescript-estree/src/convert.ts#L1842-L1843

So this line will need to be reverted

Reply:
@bradzacher What about this then? [playground](https://typescript-eslint.io/play#ts=5.4.5&showAST=es&fileType=.ts&code=MYewdgzgLgBAhjAvDAAgMxCGwA2cIQwDCMA3gL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Reply:
That case could be made a hard error - for sure!

The non-error case is correct though - [`const a = class C {}` produces an AST with `decorators: []`](https://typescript-eslint.io/play/#ts=5.4.5&showAST=es&fileType=.ts&code=MYewdgzgLgBAhjAvDYAbOEIwMIwN4C%2BQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Reply:
@bradzacher This is valid TypeScript code.